### PR TITLE
[ML] adding docs + hlrc for data frame analysis feature_processors

### PR DESCRIPTION
**Pull request by benwtrent**
Adds HLRC and some docs for the new feature_processors field in Data frame analytics.

### Preview

https://elasticsearch_61149.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/put-dfanalytics.html
https://elasticsearch_61149.docs-preview.app.elstc.co/guide/en/elasticsearch/client/java-rest/master/java-rest-high-x-pack-ml-put-data-frame-analytics.html

### CODE REVIEW

File: docs/java-rest/high-level/ml/put-data-frame-analytics.asciidoc [hunk 1]
```diff
@@ -124,6 +124,8 @@ include-tagged::{doc-tests-file}[{api}-classification]
 <10> The seed to be used by the random generator that picks which rows are used in training.
 <11> The optimization objective to target when assigning class labels. Defaults to maximize_minimum_recall.
 <12> The number of top classes to be reported in the results. Defaults to 2.
+<13> Custom feature processors that will create new features for analysis from the included document
+     fields. Note, automatic categorical feature encoding will still occur for all features.
```
Comment (lcawl) [inline docs/java-rest/high-level/ml/put-data-frame-analytics.asciidoc:128]: Not mandatory, but it would be nice to link to this page since we've got it:
```suggestion
     fields. Note, automatic categorical {ml-docs}/ml-feature-encoding.html[feature encoding] still occurs for all features.
```

File: docs/java-rest/high-level/ml/put-data-frame-analytics.asciidoc [hunk 2]
```diff
@@ -146,6 +148,8 @@ include-tagged::{doc-tests-file}[{api}-regression]
 <10> The seed to be used by the random generator that picks which rows are used in training.
 <11> The loss function used for regression. Defaults to `mse`.
 <12> An optional parameter to the loss function.
+<13> Custom feature processors that will create new features for analysis from the included document
+fields. Note, automatic categorical feature encoding will still occur for all features.
```
Comment (lcawl) [inline docs/java-rest/high-level/ml/put-data-frame-analytics.asciidoc:152]:
```suggestion
fields. Note, automatic categorical {ml-docs}/ml-feature-encoding.html[feature encoding] still occurs for all features.
```

File: docs/reference/ml/df-analytics/apis/put-dfanalytics.asciidoc [hunk 3]
```diff
@@ -438,29 +446,29 @@ PUT _ml/data_frame/analytics/model-flight-delays-pre
...
 
 <1> Source index to analyze.
-<2> This query filters out entire documents that will not be present in the 
+<2> This query filters out entire documents that will not be present in the
 destination index.
-<3> The `_source` object defines fields in the dataset that will be included or 
-excluded in the destination index. 
-<4> Defines the destination index that contains the results of the analysis and 
-the fields of the source index specified in the `_source` object. Also defines 
+<3> The `_source` object defines fields in the dataset that will be included or
```
Comment (lcawl) [inline docs/reference/ml/df-analytics/apis/put-dfanalytics.asciidoc:451]:
```suggestion
<3> The `_source` object defines fields in the data set that will be included or
```

File: docs/reference/ml/ml-shared.asciidoc [hunk 4]
```diff
@@ -522,6 +522,14 @@ The fraction of features that is used when selecting a random bag for each
 candidate split.
 end::dfas-feature-bag-fraction[]
 
+tag::dfas-feature-processors[]
+A collection of feature preprocessors that modify one or more included fields.
+The analysis uses the resulting one or more features instead of the
+original document field. Multiple `feature_processors` entries can refer to the
+same document fields.
+Note, automatic categorical feature encoding is still completed in the analysis.
```
Comment (lcawl) [inline docs/reference/ml/ml-shared.asciidoc:530]:
```suggestion
Note, automatic categorical {ml-docs}/ml-feature-encoding.html[feature encoding] still occurs.
```

**Review by lcawl — APPROVED** [review]
LGTM

File: client/rest-high-level/src/test/java/org/elasticsearch/client/ml/dataframe/ClassificationTests.java [hunk 5]
```diff
@@ -38,9 +48,21 @@ public static Classification randomClassification() {
             .setRandomizeSeed(randomBoolean() ? null : randomLong())
             .setClassAssignmentObjective(randomBoolean() ? null : randomFrom(Classification.ClassAssignmentObjective.values()))
             .setNumTopClasses(randomBoolean() ? null : randomIntBetween(0, 10))
+            .setNumTopClasses(randomBoolean() ? null : randomIntBetween(0, 10))
```
Comment (przemekwitek) [inline client/rest-high-level/src/test/java/org/elasticsearch/client/ml/dataframe/ClassificationTests.java:51]:
```suggestion
```

File: docs/reference/ml/df-analytics/apis/put-dfanalytics.asciidoc [hunk 6]
```diff
@@ -164,27 +168,27 @@ The configuration information necessary to perform
 [%collapsible%open]
 =====
 `compute_feature_influence`::::
-(Optional, boolean) 
+(Optional, boolean)
```
Comment (przemekwitek) [inline docs/reference/ml/df-analytics/apis/put-dfanalytics.asciidoc:171]: Just wondering (as my Intellij also removes trailing whitespace).
Should we piggy back the whitespace changes with our PRs or maybe there should be a separate PR to clean up the whitespace so that it is easier to review the "real" changes?

**Review by przemekwitek — APPROVED** [review]
LGTM

Just two small comments to consider.

File: docs/reference/ml/df-analytics/apis/put-dfanalytics.asciidoc (hunk 6 above)
Comment (benwtrent) [reply docs/reference/ml/df-analytics/apis/put-dfanalytics.asciidoc:171]: It is weird that trailing white spaces are there to begin with. I am not sure which is best. But backing out only the white space changes in a file would take some work/scripting.

**Comment by benwtrent** [comment]
@elasticmachine update branch
